module/os/linux/spl/spl-thread.c

```diff
@@ -158,3 +158,63 @@ spl_kthread_create(int (*func)(void *), void *data, const char namefmt[], ...)
 	} while (1);
 }
 EXPORT_SYMBOL(spl_kthread_create);
+
+/*
+ * The "why" argument indicates the allowable side-effects of the call:
+ *
+ * FORREAL:  Extract the next pending signal from p_sig into p_cursig;
+ * stop the process if a stop has been requested or if a traced signal
+ * is pending.
+ *
+ * JUSTLOOKING:  Don't stop the process, just indicate whether or not
+ * a signal might be pending (FORREAL is needed to tell for sure).
+ */
+int
+issig(int why)
+{
+	ASSERT(why == FORREAL || why == JUSTLOOKING);
+
+	if (!signal_pending(current))
+		return (0);
+
+	if (why != FORREAL)
+		return (1);
+
+	struct task_struct *task = current;
+#ifdef HAVE_SIGINFO
+	kernel_siginfo_t __info;
+#else
+	siginfo_t __info;
+#endif
```

Comment:
nit: we could make this more readable my defining our own custom type in a header.  We've done this in a few other places.

```c
#ifdef HAVE_SIGINFO
typedef kernel_siginfo_t spl_kernel_siginfo_t;
#else
typedef kersiginfo_t spl_kernel_siginfo_t;
#endif

spl_kernel_siginfo_t __info;
```